### ISSUE TITLE
As a user, I want to consume Gitcoin events via a websocket connector.

### ISSUE THREAD

**Issue by mbeacom** [gitcoinco/web#1193]
# Gitcoin Websocket Service

## TLDR

This issue is currently not ready to be worked on.  Please feel free to comment with concerns, recommendations, and general notes.  Feedback on whether or not you would find this functionality useful is also welcome and encouraged!

### User Story

As a user, I want to consume Gitcoin events via a websocket (wss) connector.

### Why Is this Needed?

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)

### Description

*Type*: Feature

### Current Behavior

No existing functionality.

### Expected Behavior

- A user should be able to connect/listen to the websocket stream via `wss`.
- `...` To be defined.

### Solution

- Add a websocket service solution like `django-channels` with `daphne`, `centrifugo`, `socket.io`, or some other websocket solution agreed upon prior to starting work. Recommended implementation: `django-channels` and an ASGI like `daphne`.
- `...` To be defined.

### Definition of Done

- PR with:
  - Mechanism for deploying the websocket service and related infrastructure
  - Requested logic/code implemented
  - All merge checks pass
  - Accepted review from Gitcoin Core member
  - Tests and docstrings for all newly introduced backend code.
  - Documentation including deployment, end-user usage, and examples.
  - Adheres to the contributor guidelines
- `WIP` not present in the PR title.
- Code has been tested locally and works with a generic websocket client.
- Code has been deployed and tested in our stage environment (https://stage.gitcoin.co)

### Additional Information

User request / Slack reference:  https://gitcoincommunity.slack.com/archives/C8E45J5D0/p1526900005000210
<img width="620" alt="screenshot 2018-05-21 09 32 29" src="https://user-images.githubusercontent.com/7315957/40310168-e709f31e-5cd9-11e8-8b06-1075884f89f3.png">

#### Reviewers

- @mbeacom, @owocki, @thelostone-mc

**Comment by owocki** [comment]
semi related: i wonder if we should move the events into a DB model called `events` and then process the slack / github / twitter / whatever notifications outside of the request/response cycle.  

celery would work too

**Comment by mbeacom** [comment]
@owocki I agree with the need for an intermediate model in the form of `Event` or `Notification`.

We could simply create the objects as necessary and queue them up for distribution/execution via celery worker.

**Comment by owocki** [comment]
oh man the opportunties to clean up `dashboard/notification.py` here 😍

**Comment by kuhnchris** [comment]
resolved with the Notification("Bell") PRs
